cmd/createtree/main.go

```diff
@@ -142,84 +126,32 @@ func newRequest(opts *createOpts) (*trillian.CreateTreeRequest, error) {
 		HashStrategy:       trillian.HashStrategy(hs),
 		HashAlgorithm:      sigpb.DigitallySigned_HashAlgorithm(ha),
 		SignatureAlgorithm: sigpb.DigitallySigned_SignatureAlgorithm(sa),
-		DisplayName:        opts.displayName,
-		Description:        opts.description,
+		DisplayName:        *displayName,
+		Description:        *description,
 		PrivateKey:         pk,
-		MaxRootDuration:    ptypes.DurationProto(opts.maxRootDuration),
+		MaxRootDuration:    ptypes.DurationProto(*maxRootDuration),
 	}}
 	return ctr, nil
 }
 
-func newPK(opts *createOpts) (*any.Any, error) {
-	switch opts.privateKeyType {
-	case "PEMKeyFile":
-		if opts.pemKeyPath == "" {
-			return nil, errors.New("empty pem_key_path")
-		}
-		if opts.pemKeyPass == "" {
-			return nil, fmt.Errorf("empty password for PEM key file %q", opts.pemKeyPath)
-		}
-		pemKey := &keyspb.PEMKeyFile{
-			Path:     opts.pemKeyPath,
-			Password: opts.pemKeyPass,
-		}
-		return ptypes.MarshalAny(pemKey)
-	case "PrivateKey":
-		if opts.pemKeyPath == "" {
-			return nil, errors.New("empty pem_key_path")
-		}
-		pemSigner, err := keys.NewFromPrivatePEMFile(
-			opts.pemKeyPath, opts.pemKeyPass)
+func newPK(format string) (*any.Any, error) {
```

Comment:
This function should probably live in it's own file or package along with the `keyHandlers` 

Also consider a clearer function name, say `NewPublicKeyProto`